src/jace/translated_jaxpr_sdfg.py

```diff
@@ -0,0 +1,229 @@
+    supports the calling convention used inside JaCe, as in `DaCe` it is callable.
+    The only valid way to obtain a `CompiledJaxprSDFG` instance is through
+    `compile_jaxpr_sdfg()`.
+
+    Args:
+        csdfg: The `CompiledSDFG` object.
+        inp_names: Names of the SDFG variables used as inputs.
+        out_names: Names of the SDFG variables used as outputs.
+
+    Attributes:
+        csdfg: The `CompiledSDFG` object.
+        sdfg: The encapsulated SDFG object.
+        inp_names: Names of the SDFG variables used as inputs.
+        out_names: Names of the SDFG variables used as outputs.
+
+    Notes:
+        Currently the strides of the input arguments must match the ones that were used
+        for lowering the SDFG.
+        In DaCe the return values are allocated on a per `CompiledSDFG` basis. Thus
+        every call to a compiled SDFG will override the value of the last call, in JaCe
+        the memory is allocated on every call. In addition scalars are returned as
+        arrays of length one.
+    """
+
+    csdfg: compiled_sdfg.CompiledSDFG
+    sdfg: dace.SDFG
+    inp_names: tuple[str, ...]
+    out_names: tuple[str, ...]
+
+    def __init__(
+        self,
+        csdfg: compiled_sdfg.CompiledSDFG,
+        inp_names: tuple[str, ...],
+        out_names: tuple[str, ...],
+    ) -> None:
+        self.csdfg = csdfg
+        self.sdfg = self.csdfg.sdfg
+        self.inp_names = inp_names
+        self.out_names = out_names
+
+    def __call__(
+        self,
+        flat_call_args: Sequence[Any],
+    ) -> list[np.ndarray] | None:
+        """
+        Run the compiled SDFG using the flattened input.
+
+        The function will not perform flattening of its input nor unflattening of
+        the output.
+
+        Args:
+            csdfg: The compiled SDFG to call.
+            flat_call_args: Flattened input arguments.
+        """
+        if len(self.inp_names) != len(flat_call_args):
+            # Either error or static arguments are not removed.
+            raise RuntimeError("Wrong number of arguments.")
+
+        sdfg_call_args: dict[str, Any] = {}
+        for in_name, in_val in zip(self.inp_names, flat_call_args):
+            # TODO(phimuell): Implement a stride matching process.
+            if util.is_jax_array(in_val):
+                if not util.is_fully_addressable(in_val):
+                    raise ValueError(f"Passed a not fully addressable Jax array as '{in_name}'")
+                in_val = in_val.__array__()  # noqa: PLW2901  # Jax arrays do not expose the __array_interface__.
+            sdfg_call_args[in_name] = in_val
+
+        arrays = self.sdfg.arrays
+        for out_name, sdfg_array in ((out_name, arrays[out_name]) for out_name in self.out_names):
+            if out_name in sdfg_call_args:
+                if util.is_jax_array(sdfg_call_args[out_name]):
+                    raise ValueError("Passed an immutable Jax array as output.")
```

Comment:
You do realize that this name, `out_name` has nothing to do with the signature of the Python function, as it has the form `__jace_input_{RANDOM_NUMBER}`, but I can do this, but it does not really tell you anything.

Reply:
But then what's the relation between `out_names` and the arguments provided by the user? Couldn't you at least say in which position was the wrong argument, if the `out_names` tuple somehow preservers the order of some user-provided arguments?

Reply:
Order is also not really applicable, because:
- A pytree will change that for example arguments `ndarray, tuple[ndarray, ndarray], JaxArray` will become `ndarray, ndarray, ndarray, JaxArray`, so it is actually the third argument, but it will be seen as forth.
- If the same as above happens, but one is inside a pytree.
- If you pass them as keyword arguments.